mitiq/shadows/quantum_processing.py

```diff
@@ -0,0 +1,183 @@
+from typing import Tuple, Callable, Dict, Any, List, Union
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+from qiskit_aer import Aer
+from tqdm.auto import tqdm
```

Comment:
I haven't had a chance to look at this PR in full yet, but if you want to add a new library as a dependency, you'll have to add it to our `requirements.txt` or `dev_requirements.txt` file so this information will trickle down to users when doing installs.

Reply:
have put tqdm and Aer as optional imports.
P.S. qiskit_aer is included in qiskit installation, which is already in dev_requirements.txt
https://github.com/Min-Li/mitiq/blob/990145c9b8c591a53f64b2aff8699a90f6f2cfc1/mitiq/shadows/quantum_processing.py#L7-L14